.github/workflows/wait-for-status-checks.yml

```diff
@@ -0,0 +1,42 @@
+name: wait-for-status-checks
+on:
+  pull_request:
+  status: {}
+jobs:
+  enforce-all-checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - name: Use correct sha
+        run: |
+          echo "before: $GITHUB_SHA"
+          if [ "$GITHUB_EVENT_NAME" == "pull_request" ]; then
+           GITHUB_SHA=${{ github.event.pull_request.head.sha }}
+          fi
+          echo COMMIT_SHA=$GITHUB_SHA >> $GITHUB_ENV
+          echo "after: $GITHUB_SHA"
```

Comment:
Tu sais pourquoi ce n'est pas le bon sha au départ ?

Reply:
[Github dans la doc ](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request) dit que la valeur de `GITHUB_SHA` quand c'est un évènement de type `pull_request` équivaut au sha du commit de merge, seulement ce n'est pas le commit de merge qui a les status mais bien le dernier commit de la branche. 

![Screenshot 2024-09-23 at 14 45 09](https://github.com/user-attachments/assets/5b9e9baf-2b13-4b48-8005-cf0de86cfd33)